src/lib/arch/aarch64/Instruction_decode.cc

```diff
@@ -208,186 +225,140 @@ void Instruction::decode() {
-            // destinations
-            destinationRegisters_[destinationRegisterCount_] =
-                csRegToRegister(op.reg);
-            destinationRegisterCount_++;
-          }
+          // Add register writes to destinations, but skip zero-register
+          // destinations
+          destinationRegisters_[destinationRegisterCount_] =
+              csRegToRegister(op.reg);
+          destinationRegisterCount_++;
         }
       }
       if (op.access & cs_ac_type::CS_AC_READ) {
-        if ((op.reg >= ARM64_REG_ZAB0 && op.reg < ARM64_REG_V0) ||
-            (op.reg == ARM64_REG_ZA)) {
-          // Add all Matrix register rows as source operands
-          std::vector<Register> regs =
-              getZARowVectors(op.reg, architecture_.getStreamingVectorLength());
-          // Update source operand structure sizes
-          sourceRegisters_.addSMEOperand(regs.size());
-          sourceValues_.addSMEOperand(regs.size());
-          for (size_t i = 0; i < regs.size(); i++) {
-            sourceRegisters_[sourceRegisterCount_] = regs[i];
-            sourceRegisterCount_++;
-            sourceOperandsPending_++;
-          }
-        } else {
-          // Add register reads to destinations
-          sourceRegisters_[sourceRegisterCount_] = csRegToRegister(op.reg);
-          sourceRegisterCount_++;
-          sourceOperandsPending_++;
-        }
-        // TODO checking of the shift type is a temporary fix to help reduce the
-        // chance of incorrectly reverted aliases from being mis-classified as
-        // isShift when op.shift contains garbage data. This should be reviewed
-        // on the next capstone update which should remove the need to revert
-        // aliasing
-        if (op.shift.type > arm64_shifter::ARM64_SFT_INVALID &&
-            op.shift.type <= arm64_shifter::ARM64_SFT_ROR &&
+        // Add register reads to destinations
+        sourceRegisters_[sourceRegisterCount_] = csRegToRegister(op.reg);
+        sourceRegisterCount_++;
+        sourceOperandsPending_++;
+
+        // Identify shift operands
+        if (op.shift.type != aarch64_shifter::AARCH64_SFT_INVALID &&
             op.shift.value > 0) {
-          setInstructionType(InsnType::isShift);  // Identify shift operands
+          setInstructionType(InsnType::isShift);
         }
       }
-    } else if (op.type == ARM64_OP_MEM) {  // Memory operand
-      accessesMemory = true;
-      sourceRegisters_[sourceRegisterCount_] = csRegToRegister(op.mem.base);
-      sourceRegisterCount_++;
-      sourceOperandsPending_++;
-
-      if (metadata_.writeback) {
-        // Writeback instructions modify the base address
-        destinationRegisters_[destinationRegisterCount_] =
-            csRegToRegister(op.mem.base);
-        destinationRegisterCount_++;
+    } else if (op.type == AARCH64_OP_MEM) {  // Memory operand
+      // Check base register exists
+      if (op.mem.base != AARCH64_REG_INVALID) {
+        accessesMemory = true;
```

Comment:
Do we have instructions that have `AARCH64_OP_MEM` ops with no base register?

Reply:
This is a safeguard against one of the un-used operands (as the Capstone array is fixed at 16 entries) having the type of `AARCH64_OP_MEM` but then no register. I think it came up once with one instruction hence the addition